docs/dymos_book/examples/balanced_field/balanced_field_funccomp.ipynb

```diff
@@ -0,0 +1,590 @@
+    "    K_nom = 1.0 / (np.pi * AR * e)\n",
+    "    b = span / 2.0\n",
+    "    fact = ((h + h_w) / b) ** 1.5\n",
+    "    K = K_nom * 33 * fact / (1.0 + 33 * fact)\n",
+    "\n",
+    "    q = 0.5 * rho * v ** 2\n",
+    "    L = q * S * CL\n",
+    "    D = q * S * (CD0 + K * CL ** 2)\n",
+    "\n",
+    "    # Compute the downward force on the landing gear\n",
+    "    calpha = np.cos(alpha)\n",
+    "    salpha = np.sin(alpha)\n",
+    "\n",
+    "    # Runway normal force\n",
+    "    F_r = m * g - L * calpha - T * salpha\n",
+    "\n",
+    "    # Compute the dynamics\n",
+    "    v_dot = (T * calpha - D - F_r * mu_r) / m\n",
+    "    r_dot = v\n",
+    "\n",
+    "    return CL, q, L, D, K, F_r, v_dot, r_dot, W, v_stall, v_over_v_stall"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Climb ODE function"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def climb_ode(rho, S, CD0, CL0, CL_max, alpha_max, h_w, AR, e, span, T, m, v, h, alpha, gam):\n",
+    "    g = 9.80665\n",
+    "    W = m * g\n",
+    "    v_stall = np.sqrt(2 * W / rho / S / CL_max)\n",
+    "    v_over_v_stall = v / v_stall\n",
+    "\n",
+    "    CL = CL0 + (alpha / alpha_max) * (CL_max - CL0)\n",
+    "    K_nom = 1.0 / (np.pi * AR * e)\n",
+    "    b = span / 2.0\n",
+    "    fact = ((h + h_w) / b) ** 1.5\n",
+    "    K = K_nom * 33 * fact / (1.0 + 33 * fact)\n",
+    "\n",
+    "    q = 0.5 * rho * v ** 2\n",
+    "    L = q * S * CL\n",
+    "    D = q * S * (CD0 + K * CL ** 2)\n",
+    "\n",
+    "    # Compute the downward force on the landing gear\n",
+    "    calpha = np.cos(alpha)\n",
+    "    salpha = np.sin(alpha)\n",
+    "\n",
+    "    # Runway normal force\n",
+    "    F_r = m * g - L * calpha - T * salpha\n",
+    "\n",
+    "    # Compute the dynamics\n",
+    "    cgam = np.cos(gam)\n",
+    "    sgam = np.sin(gam)\n",
+    "    v_dot = (T * calpha - D) / m - g * sgam\n",
+    "    gam_dot = (T * salpha + L) / (m * v) - (g / v) * cgam\n",
+    "    h_dot = v * sgam\n",
+    "    r_dot = v * cgam\n",
+    "\n",
+    "    return CL, q, L, D, K, F_r, v_dot, r_dot, W, v_stall, v_over_v_stall, gam_dot, h_dot"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### The Function Wrapping Component\n",
+    "\n",
+    "The following component wraps one of the above ODE functions depending on its 'mode'.\n",
+    "This allows us to declare a single ODE component for use in the problem regardless of whether the given phase models the ground roll or the climb."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import openmdao.api as om\n",
+    "import openmdao.func_api as omf\n",
+    "\n",
+    "\n",
+    "class BalancedFieldODEComp(om.ExplicitFuncComp):\n",
```

Comment:
I would not sub-class ExplicitFuncComp here. I would make a contructor function that builds the meta object for you, with some arguments like num_nodes, and other switches. Then just pass that meta object to the stock-standard ExplicitFunctionComp. 

Hopefully the function-comp api gives you a way to declare an option somehow... and you can use that option when you call the meta constructor inside the phase's setup. The component can have that option, even though non of the compute methods need it. You might still need to pass it in to the function though.

Reply:
@JustinSGray Implemented as a factory function instead of subclassing.  I agree it's cleaner.